src/WebAPI.jl

```diff
@@ -0,0 +1,307 @@
+        notebook = "..."
+        if running
+            return HTTP.Response(200, "Notebook already running, nothing to do!")
+        end
+        path, io = mktemp()
+        notebookbytes = HTTP.payload(request)
+        @info path typeof(notebookbytes), notebookbytes 
+        hash = myhash(notebookbytes)
+        open(path, "w") do io
+            write(io, notebookbytes)
+        end
+        add_to_session!(server_session, notebook_sessions, path)
+
+        HTTP.Response(200, "Started notebook successfully " * hash)
+    end
+
+    function stop_notebook(request::HTTP.Request)
+        sesh = get_sesh(request)
+        notfound = sesh === nothing
+        running = sesh isa RunningNotebookSession
+        queued = sesh isa QueuedNotebookSession
+        finished = sesh isa FinishedNotebookSession
+        notebook_hash = sesh.hash
+        if running
+            remove_from_session!(server_session, notebook_sessions, notebook_hash)
+            msg = "Successfully shut down " * sesh.hash
+        else
+            msg = "Nothing to do!"
+        end
+        @info msg
+        return HTTP.Response(200, msg)
+    end
+
+    function reload_filesystem(request::HTTP.Request)
+        
+        if get(ENV, "GITHUB_SECRET", "") !== ""
+            i = findfirst(a -> lowercase(a.first) == lowercase("X-Hub-Signature-256"), request.headers)
+            @info request.headers i
+            if (isnothing(i))
+                @warn "Can't validate: header not found"
+            end
+            secure_header = request.headers[i].second
+            digest = "sha256=" * bytes2hex(hmac_sha256(collect(codeunits(ENV["GITHUB_SECRET"])), request.body))
+            println(length(request.body))
+            println("Digest: " * digest)
+            println("header: " * secure_header)
+            security_test = digest == secure_header
+            if !security_test
+                return HTTP.Response(501, "Not authorized!")
+            end
+        end
+        
+        params = HTTP.queryparams(HTTP.URI(request.target))
+        github_url = get(get(JSON.parse(String(request.body)), "repository", Dict()), "html_url", nothing)
+        folder = !isnothing(github_url) ? split(github_url, "/")[end] : "spam"
+        exclude_hases = get(params, "exclude", [])
+        @async try
+            if length(folder) > 0 
+                toclone = github_url
+                this_folder = pwd()
+                @info this_folder
+                run(`rm -rf "$folder"`)
+                # Clone without history
+                # Fetch/Pull if you have latest
+                # Also have some cleanup around!
+                run(`git clone "$toclone"`)
+            else 
+                return HTTP.Response(501, "Can't pull")
+            end
+            start_dir = "$this_folder/$folder"
+            settings, pluto_options = get_configuration("$this_folder/$folder/pluto-deployment-environment/PlutoDeployment.toml")
```

Comment:
if that is different, restart 
write a file (.env~!!!~~~) that the process will find after boot
and run `shutdown()`